Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,893 @@
+                Write-Host ""
+            }
+        }
+
+        $needsRefresh = 0
+        if ($EnableAMSI) {
+            if ($filterList) {
+                foreach ($server in $filterList) {
+                    Write-Host $bar
+                    Write-Host ""
+                    $getSOs = $null
+                    $getSOs = Get-SettingOverride -ErrorAction SilentlyContinue | Where-Object {
+                        ($_.ComponentName.ToLower() -eq 'Cafe'.ToLower()) -and
+                        ($_.SectionName.ToLower() -eq 'HttpRequestFiltering'.ToLower()) -and
+                        ($_.Parameters.ToLower() -eq 'Enabled=False'.ToLower()) -and
+                        ($null -ne $_.Server -and ($_.Server.ToLower() -contains $server.ToLower())) }
+                    if ($null -eq $getSOs) {
+                        Write-Host "We did not find Get-SettingOverride that disabled AMSI on $server"
+                        Write-Warning "AMSI is NOT disabled on $server"
+                    } else {
+                        foreach ($so in $getSOs) {
+                            if ($so.Status -eq 'Accepted') {
+                                Write-Warning "AMSI is Disabled by $($so.Name) SettingOverride on $server"
+                            } else {
+                                Write-Warning "AMSI is Disabled by $($so.Name) SettingOverride on $server but it is not Accepted."
+                            }
+                            $so | Out-Host
+                            if (-not $WhatIfPreference) { Write-Host "Removing SettingOverride $($so.Name)" }
+                            $rso = $null
+                            Remove-SettingOverride -Identity $so.Identity -Confirm:$false -WhatIf:$WhatIfPreference -ErrorVariable $rso
+                            if (-not $WhatIfPreference) {
+                                if ($rso) {
+                                    Write-Host "We could not remove the SettingOverride on $server" -ForegroundColor Red
+                                } else {
+                                    Write-Host "Enabled on $server" -ForegroundColor Green
+                                    Get-ExchangeDiagnosticInfo -Process Microsoft.Exchange.Directory.TopologyService -Component VariantConfiguration -Argument Refresh -Server $server | Out-Null
+                                    Write-Host "Refreshed Get-ExchangeDiagnosticInfo on $server"
+                                    $needsRefresh++
+                                }
+                            }
+                        }
+                    }
+                }
+            } else {
+                Write-Host $bar
+                Write-Host ""
+                $getSOs = $null
+                $getSOs = Get-SettingOverride -ErrorAction SilentlyContinue | Where-Object {
+                    ($_.ComponentName.ToLower() -eq 'Cafe'.ToLower()) -and
+                    ($_.SectionName.ToLower() -eq 'HttpRequestFiltering'.ToLower()) -and
```

Comment:
It would be better to not use `.ToLower()` for string comparison (see: https://learn.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings#recommendations-for-string-usage). I've also learned this recently as it could lead to F/P when using specific cultures (e.g.tr-TR). Better would be using `$string.Equals("StringToCompare", [System.StringComparison]::OrdinalIgnoreCase)`